bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/LegalEntityService.kt

```diff
@@ -40,11 +43,17 @@ class LegalEntityService(
     private val outputSaasMappingService: OutputSaasMappingService,
     private val saasClient: SaasClient,
     private val poolClient: PoolClient,
+    private val changelogRepository: ChangelogRepository
 ) {
 
     private val logger = KotlinLogging.logger { }
 
     fun upsertLegalEntities(legalEntities: Collection<LegalEntityGateInputRequest>) {
+
+        legalEntities.forEach { legalEntity ->
+            changelogRepository.save(ChangelogEntity(legalEntity.externalId, LsaType.LegalEntity))
+        }
```

Comment:
Better move the changelog after the actual upsert since the call to CDQ could go wrong